rai/node/bootstrap.cpp

```diff
@@ -1022,6 +1024,19 @@ void rai::bootstrap_attempt::requeue_pull (rai::pull_info const & pull_a)
 		pulls.push_front (pull);
 		condition.notify_all ();
 	}
+	else if (pull.attempts == 4 && !connection_frontier_request.expired ())
+	{
+		std::lock_guard<std::mutex> lock (mutex);
+		auto connection_shared = connection_frontier_request.lock ();
```

Comment:
I believe the  `weak_ptr#lock` result should be checked against nullptr here, since *if (!expired) + mutex lock* isn't atomic. So it may technically have expired at this point?